### ISSUE TITLE
linter: False positive in `unicorn/consistent-function-scoping`

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#12513]
### What version of Oxlint are you using?

0.78.0

### What does your `.oxlintrc.json` config file look like?

n/a

### What happened?

In this case, `inner` does contain a reference to a variable outside of its scope:

```js
function outer() {
  {
    let x;
    var inner = () => x;
  }
  return inner;
}
```

Linter incorrectly produces an error:

```
⚠ eslint-plugin-unicorn(consistent-function-scoping): Function `inner` does not capture any variables from its parent scope
   ╭─[consistent_function_scoping.tsx:4:25]
 1 │ function outer() {
   ·          ──┬──
   ·            ╰── Outer scope where this function is defined
 2 │                 {
 3 │                     let x;
 4 │                     var inner = () => x;
   ·                         ──┬──
   ·                           ╰── This function does not use any variables from the parent function
 5 │                 }
   ╰────
  help: Move `inner` to the outer scope to avoid recreating it on every call.
```

Looks like problem is that the rule determines what scopes are outside of the function by starting with the scope containing the variable binding `inner`. In this case, it's a `var` statement, so that binding is hoisted up to a scope above the function itself.

I'm not sure why the rule works like this. On the face of it, it'd be simpler to skip all the logic of finding the `VarDeclarator` binding, and just create a list of parent scopes by starting from the function's own scope. That'd also fix this problem, and also handle cases where it's hard to locate the relevant `VarDeclarator`, like this:

```js
const [ { inner } ] = [ { inner: () => x } ];
```

But I may be missing something...
